packages/admin/cms-admin/src/pages/pageTree/useExtractPages.tsx

```diff
@@ -0,0 +1,270 @@
+                                                        ? transformToSlug(importedSlug ?? page.slug, locale)
+                                                        : page.slug,
+                                                attachedDocument: { id: data.page.document.id, type: data.page.document.__typename },
+                                            },
+                                        },
+                                        context: LocalErrorScopeApolloContext,
+                                    });
+                                }
+                            }
+                        }
+                    } catch (e) {
+                        throw new Error(`Error while fetching page`);
+                    }
+                }),
+            );
+        },
+        [client, documentTypes, locale],
+    );
+
+    const getContentsFromClipboard = async (): Promise<GetContentsFromClipboardResponse> => {
+        const text = await readClipboard();
+
+        if (text === undefined) {
+            return {
+                canPaste: false,
+                error: (
+                    <FormattedMessage
+                        id="comet.pages.cannotPastePage.messageFailedToReadClipboard"
+                        defaultMessage="Can't read clipboard content. Please make sure that clipboard access is given"
+                    />
+                ),
+            };
+        }
+
+        if (text.trim() === "") {
+            return {
+                canPaste: false,
+                error: <FormattedMessage id="comet.pages.cannotPastePage.messageEmptyClipboard" defaultMessage="Clipboard is empty" />,
+            };
+        }
+
+        try {
+            const parsedText = JSON.parse(text);
+            if (isContentsClipboard(parsedText)) {
+                return {
+                    canPaste: true,
+                    content: {
+                        contents: Object.entries(parsedText.textContents).map((content) => ({ original: content[0], replaceWith: content[1] })),
+                    },
+                };
+            } else {
+                return {
+                    canPaste: false,
+                    error: (
+                        <FormattedMessage
+                            id="comet.pages.cannotPasteBlock.messageFailedToParseClipboard"
+                            defaultMessage="Content from clipboard aren't valid blocks"
+                        />
+                    ),
+                };
+            }
+        } catch (e) {
+            return {
+                canPaste: false,
+                error: (
+                    <FormattedMessage
+                        id="comet.pages.cannotPasteBlock.messageFailedToParseClipboard"
+                        defaultMessage="Content from clipboard aren't valid blocks"
+                    />
+                ),
+            };
+        }
+    };
```

Comment:
These error messages are from the block clipboard. Please adapt them for contents import/export